library/constant_time_impl.h

```diff
@@ -0,0 +1,306 @@
+/**
+ *  Constant-time functions
+ *
+ *  For readability, the static inline definitions are here, and
+ *  constant_time_internal.h has only the declarations.
+ *
+ *  This results in duplicate declarations of the form:
+ *      static inline void f() { ... }
+ *      static inline void f();
```

Comment:
(minor; can be fixed in a later PR) this is now the other way round, right? We `#include "constant_time_impl.h"` at the end of `constant_time_internal.h`.

And "behave exactly as if" isn't quite right - we have warnings to silence; we do them in this file to hide that from `constant_time_internal.h`

Reply:
Covered in https://github.com/Mbed-TLS/mbedtls/pull/7590/commits/be5676a2e1ce85021985de8033a87992345a93d1